pyxdf/pyxdf.py

```diff
@@ -702,6 +703,8 @@ def _robust_fit(A, y, rho=1, iters=1000):
     http://www.stanford.edu/~boyd/papers/distr_opt_stat_learning_admm.html
 
     """
+    offset = np.min(A[:, 1])
+    A[:, 1] = A[:, 1] - offset  # Don't use -= operator here!
```

Comment:
Why not `-=`?

```
In [1]: import numpy as np

In [2]: A=np.random.rand(5,2)

In [3]: A
Out[3]: 
array([[0.894628  , 0.03031772],
       [0.15520362, 0.78182106],
       [0.02175627, 0.70150644],
       [0.42385584, 0.4072398 ],
       [0.77578102, 0.10038374]])

In [4]: A[:,1]-=np.min(A[:,1])

In [5]: A
Out[5]: 
array([[0.894628  , 0.        ],
       [0.15520362, 0.75150334],
       [0.02175627, 0.67118871],
       [0.42385584, 0.37692208],
       [0.77578102, 0.07006602]])
```

Reply:
Probably because it operates on a slice? https://stackoverflow.com/a/9047211/1112283
In any case, the comment should explain why `-=` should not be used!

Reply:
I was trying to not assume that the second column of A was not shared somewhere else so I didn't want to mutate data in-place. But, this is a bit silly because I am still making many other assumptions that are much more likely to be invalidated than the above assumption. For example, I'm assuming that A will never be reused so it's ok to mutate its second column.

The better solution is to first copy A before modifying it. This is now updated.